lib/simctl.js

```diff
@@ -1,21 +1,28 @@
 import Q from 'q';
-import cp from 'child_process';
+import { exec } from 'teen_process';
 import { sleep, retry, nodeifyAll } from 'asyncbox';
 import npmlog from 'npmlog';
 import { mapify } from 'es6-mapify';
 
 let log = global._global_npmlog || npmlog;
 
-function simExec (cmd:string, timeout:number, args:Array = []) {
-  args = args.map((arg) => {
-    if (arg.indexOf(" ") !== -1) {
-      return '"' + arg + '"';
+function simExec (command:string, timeout:number, args:Array = []) {
+
+  args = ["simctl", command].concat(args);
+  log.info(`Executing: xcrun with args: ${args.join(' ')} and timeout: ${timeout}`);
+
+  let ret;
+  try {
+    ret = exec("xcrun", args, {timeout: timeout});
+  } catch (e) {
+    if (e.stderr) {
+      throw new Error(`sim-ctl error: ${e.stderr.trim()}`);
+    } else {
+      throw e;
     }
```

Comment:
@sebv how is this error handling? If e.stderr exists it must be an instruments error so we print stderr. otherwise it could be an exec error so we just bubble it up.